app/models/user.py

```diff
@@ -0,0 +1,214 @@
+import re
+import uuid
+from typing import Optional
+
+from pydantic import BaseModel, validator
+from sqlmodel import Field, Relationship, SQLModel
+
+
+class DiscordModel(SQLModel, table=True):
+    id: Optional[int] = Field(default=None, primary_key=True)
+    email: Optional[str] = None
+    mfa: Optional[bool] = None
+    avatar: Optional[str] = None
+    banner: Optional[str] = None
+    color: Optional[int] = None
+    nitro: Optional[int] = None
+    locale: Optional[str] = None
+    username: str
+
+    user_id: Optional[uuid.UUID] = Field(default=None, foreign_key="usermodel.id")
+    user: "UserModel" = Relationship(back_populates="discord")
+
+
+class EthicsFormModel(SQLModel, table=True):
+    id: Optional[int] = Field(default=None, primary_key=True)
+    hack_others: Optional[bool] = False
+    hack_ucf: Optional[bool] = False
+    interrupt_ucf: Optional[bool] = False
+    manip_traffic: Optional[bool] = False
+    bypass_dhcp: Optional[bool] = False
+    pirate: Optional[bool] = False
+    host_at_ucf: Optional[bool] = False
+    signtime: Optional[int] = 0
+
+    user_id: Optional[uuid.UUID] = Field(default=None, foreign_key="usermodel.id")
+    user: "UserModel" = Relationship(back_populates="ethics_form")
+
+
+
+class UserModel(SQLModel, table=True):
```

Comment:
Should almost all of these fields be nullable?

Reply:
I will review, but most of the values are unknown when we first write the user to the DB before they do the onboarding forms